src/main/java/foundation/stack/jdbc/NameGenerator.java

```diff
@@ -16,19 +16,12 @@
  * @author Ravi Chodavarapu (rchodava@gmail.com)
  */
 public class NameGenerator {
-    private static final String JDBC_DRIVER_PACKAGE = DockerDatabaseServerPerApplicationConnectionLookup.class.getPackage().getName();
 
     private static String getCallerClassName() {
         StackTraceElement[] stackTrace = Thread.currentThread().getStackTrace();
         if (stackTrace != null && stackTrace.length > 0) {
-            for (int i = 1; i < stackTrace.length; i++) {
-                StackTraceElement element = stackTrace[i];
-                String className = element.getClassName();
-
-                if (!className.startsWith(JDBC_DRIVER_PACKAGE)) {
-                    return className;
-                }
-            }
+            int index = stackTrace.length == 1 ? 0 : stackTrace.length - 1;
```

Comment:
Not sure about this statement. The reason why it was changed was because it was picking up tomcat jar as the first in the stack not matching package. I believe a similar scenario will happen no matter what, as the driver will always be initialised by some framework (this is rarely done by application user code).
Regarding the bottom of the stack, for applications ran from a user provided class's main (i.e. spring-boot): when running from IDE it should always be able to find the git location, when running from a jar, it actually uses "java.lang.Thread" as the name.